spec/features/update_course_stats_average_views_importer_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe UpdateCourseStats do
+  let(:course) { create(:course) }
+  let!(:article1) { create(:article) }
+  let!(:article2) { create(:article) }
+  let(:update_service) { described_class.new(course) }
+
+  before do
+    course.articles << [article1, article2]
+  end
+
+  describe '#update_average_pageviews' do
+    it 'updates outdated average views for the course articles' do
+      expect(AverageViewsImporter).to receive(:update_outdated_average_views)
+        .with(course.articles, update_service:)
+
+      # Call the update_average_pageviews method directly
+      update_service.send(:update_average_pageviews)
+    end
+
+    it 'logs the progress of average pageviews updates' do
```

Comment:
The important aspect of the update service is to collect and log errors that happen during the update, (such as when an API is unavailable or overloaded). Have you tested that functionality?